client-haskell/src/Icepeak/Client.hs

```diff
@@ -48,9 +53,14 @@ import qualified Network.HTTP.Client as HTTP
 import qualified Network.HTTP.Types.Status as HTTP
 import qualified Network.HTTP.Types.Header as Header
 import qualified Network.HTTP.Types.URI as URI
+import qualified Control.Retry as Retry
 
 -- | Config paired with HTTP manager for making requests.
-data Client = Client HTTP.Manager Config
+data Client m = Client 
+  { clientHttpManager :: HTTP.Manager
+  , clientConfig      :: Config
+  , clientRetryPolicy :: Maybe (RetryPolicyM m)
```

Comment:
Adding a new field here breaks the integration tests:

https://github.com/channable/icepeak/blob/fb30e95429010f1d46a29ca312724437da5ac596/server/integration-tests/stress_test.hs#L36

https://github.com/channable/icepeak/blob/fb30e95429010f1d46a29ca312724437da5ac596/server/integration-tests/auth_stress_test.hs#L39

Reply:
Indeed, missed this.